lib/src/base-styles/_nx-forms.scss

```diff
@@ -10,88 +10,74 @@
 
 .nx-form {
   @include container-vertical;
-  border-top: $nx-border-dark;
   margin: 0;
-  padding-top: 20px;
   position: relative; // to make a containing block for validation popovers
 }
 
-.nx-form--simple {
-  border-top: none;
-  padding: 0;
-}
+.nx-form-footer {
+  @include container-horizontal;
 
-// for a form that is the entirety of a modal
-.nx-form--modal-form {
-  padding-top: 0;
+  border-top: 1px solid #e9e9e9;
+  padding: $nx-spacing-xl $nx-spacing-l;
+  text-align: right;
 }
 
 .nx-form-group {
-  margin-bottom: 20px;
+  margin-bottom: 6px;
```

Comment:
This kinda sucks, but the spacing between groups must take into account the validation message space below the text inputs.

Reply:
> This kinda sucks, but the spacing between groups must take into account the validation message space below the text inputs.

Or should the specs be changed such that this can become a more standard 8px?

Reply:
> Or should the specs be changed such that this can become a more standard 8px?

Ah but then we couldn't use the standard 32px margin on the bottom of the fieldsets.  I think the root of the problem is that the validation message is 18px tall (12px font size * 1.5 line-height).  2px too tall to fit cleanly into our multiples-of-8 spacing scheme.  We could force the line-height to 16px there.

Reply:
> > Or should the specs be changed such that this can become a more standard 8px?
> 
> Ah but then we couldn't use the standard 32px margin on the bottom of the fieldsets. I think the root of the problem is that the validation message is 18px tall (12px font size * 1.5 line-height). 2px too tall to fit cleanly into our multiples-of-8 spacing scheme. We could force the line-height to 16px there.

The validation message uses a standard font' and margin:
```
.nx-text-input__invalid-message {
  @include regular();
  color: #cd0028;
  font-size: $nx-font-size-xs;
  margin-top: $nx-spacing-xs;
}
```
So if we customize that we're really just moving the magic number from here to there.

Reply:
> So if we customize that we're really just moving the magic number from here to there.

A magic number will be necessary somewhere for this.  I think I prefer it being the line-height that is custom rather than the margin.

Reply:
> > So if we customize that we're really just moving the magic number from here to there.
> 
> A magic number will be necessary somewhere for this. I think I prefer it being the line-height that is custom rather than the margin.

I think that makes more sense: https://github.com/sonatype/sonatype-react-shared-components/pull/122/commits/4bd611669fd5e61a800a3b345de9c967e56274eb